src/app/Library/CrudPanel/Traits/Query.php

```diff
@@ -151,4 +152,150 @@ public function orderByWithPrefix($column_name, $column_direction = 'ASC')
+     *
+     * For example a base query of: `SELECT * FROM table WHERE (someColumn, smtValue)`, we would return only the `someColumn`
+     * with the objective of replacing the `*` for the specific columns needed, avoiding the selection of
+     * columns that would not have impact in the counting process.
+     *
+     * @param  \Illuminate\Database\Eloquent\Builder  $query
+     * @param  bool  $nested  used to prevent multiple level nesting as we only need the first level columns
+     * @return array
+     */
+    private function getQueryColumnsFromWheres($query, $nested = false)
+    {
+        // if there is a raw where we are better not touching the columns, otherwise we
+        // would need to parse the raw sql wheres and that can get messy very quick.
+        if (in_array('raw', array_column($query->wheres, 'type'))) {
+            return $query->columns;
+        }
+
+        $wheresColumns = [];
+        foreach ($query->wheres as $where) {
+            switch ($where['type']) {
+                // case it's a basic where, we just want to select that column.
+                case 'Basic':
+                    $wheresColumns[] = $where['column'];
+                break;
+                // when it's a nested query we will get the columns that link
+                // to the main table from the nested query wheres.
+                case 'Nested':
+                    $wheresColumns = $nested ?: array_merge($wheresColumns, $this->getNestedQueryColumns($where['query']));
+                break;
+                // when Column get the "first" key that represent the base table column to link with
+                case 'Column':
+                    $wheresColumns[] = $where['first'];
+                break;
+                // in case of Exists, we will find in the subquery the query type Column where it links to the main table
+                case 'Exists':
+                    $wheres = $where['query']->wheres;
+                    foreach ($wheres as $subWhere) {
+                        if ($subWhere['type'] === 'Column') {
+                            $wheresColumns[] = $subWhere['first'];
+                        }
+                    }
+                break;
+            }
+        }
+
+        return $wheresColumns;
+    }
+
+    /**
+     * Get the query count without any filters or search applied.
+     *
+     * @return int
+     */
+    public function getTotalEntryCount()
+    {
+        if (! $this->getOperationSetting('showEntryCount')) {
+            return 0;
+        }
+
+        return  $this->getRequest()->input('totalEntryCount') ??
+                $this->getOperationSetting('totalEntryCount') ??
+                $this->getCurrentEntryCount();
+    }
+
+    /**
+     * Return the current query count.
+     *
+     * @return int
+     */
+    public function getCurrentEntryCount()
+    {
+        return $this->getEntryCount();
+    }
```

Comment:
I see this method doesn't really do anything other than call a private method. Soo... do we really need it?

Reply:
The reason for this acessor method is for keeping `getEntryCount` private, and allow `ListOperation` to call it.

Reply:
I understand that. And I would have liked for `getEntryCount()` to be `private` - would have allowed us to change it later. 

But if the only thing a `public` method does is call a `private` method... then that `private` method isn't actually safe at all... it's indirectly callable. Making it `private` and creating an alias for it... doesn't help at all. At least I don't see how.

Reply:
So my question then becomes... why does `ListOperation` need to call this method directly? `private`/`public`, doesn't matter. Why doesn't `ListOperation` call `getTotalEntryCount()` which
- (1) is already `public`
- (2) loads the count from cache if it exists

?

Reply:
I kind of agree, maybe `protected` would be the best fit here then ? So it could be called by descendants and parents of the class, but not from "outsiders" of the class ? 

```php
class MyPanel extends CrudPanel {
// private getEntryCount can NOT be called
// protected getEntryCount can be called
}

//outsiders
$x = new CrudPanel();
$x->getEntryCount() // will fail in protected and private
```

I think in the end this is a public method as it is a method that is part of the public api (called by list operation, but can be called in other places), I just wanted to put it behind some sketchy visibility setting so our lifes get's easier if we ever neeed to change the method.

Reply:
> So my question then becomes... why does `ListOperation` need to call this method directly? `private`/`public`, doesn't matter. Why doesn't `ListOperation` call `getTotalEntryCount()` which
> 
> * (1) is already `public`
> * (2) loads the count from cache if it exists
> 
> ?

`getCurrentEntryCount` it's not the same as `getTotalEntryCount`. I am getting confused by the naming, let me settle the ideas and I will check back here later.

Reply:
@tabacitu heads up, I did some changes.
- Removed request from the CrudPanel and pass it as a parameter to the function.
- renamed `getCurrentEntryCount` to `performQueryEntryCount`. 

I think this renaming, even if it's not the best name yet, may better describe what is happening in the `search()` function lifecycle. 

I think the name was misleading on what was happening, it made sense only in my head I guess. 

Please review @tabacitu  and let me know if it's now more clear of what's happening and why we don't call `getTotalEntryCount`.

Cheers